tests/networkpolicy_test.go

```diff
@@ -176,64 +239,150 @@ func skipNetworkPolicyRunningOnKindInfra() {
 	}
 }
 
-func createVMICirros(virtClient kubecli.KubevirtClient, namespace string, labels map[string]string) *v1.VirtualMachineInstance {
+func createVMICirros(virtClient kubecli.KubevirtClient, namespace string, labels map[string]string, opts ...libvmi.Option) *v1.VirtualMachineInstance {
 	var err error
 	vmi := tests.NewRandomVMIWithEphemeralDiskAndUserdata(cd.ContainerDiskFor(cd.ContainerDiskCirros), "#!/bin/bash\necho 'hello'\n")
 	vmi.Namespace = namespace
 	vmi.Labels = labels
+
+	// Clean up interfaces since we configure them with `libvmi.Option`
+	vmi.Spec.Domain.Devices.Interfaces = []v1.Interface{}
+
+	for _, opt := range opts {
+		opt(vmi)
+	}
+
 	vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Create(vmi)
 	Expect(err).ToNot(HaveOccurred())
 
 	return vmi
 }
 
-func assertPingSucceed(fromVmi, toVmi *v1.VirtualMachineInstance) {
-	ExpectWithOffset(1, toVmi.Status.Interfaces[0].IPs).NotTo(BeEmpty())
-	ConsistentlyWithOffset(1, func() error {
-		for _, toIp := range toVmi.Status.Interfaces[0].IPs {
-			if err := tests.PingFromVMConsole(fromVmi, toIp); err != nil {
-				return err
-			}
-		}
-		return nil
-	}, 15*time.Second, 1*time.Second).Should(Succeed())
+func assertICMPPingSucceed(fromVmi, toVmi *v1.VirtualMachineInstance) {
+	assertIPConnectivitySucceed(toVmi, checkICMPPingAndStopOnFailure(fromVmi))
 }
 
-func assertPingFail(fromVmi, toVmi *v1.VirtualMachineInstance) {
-	ExpectWithOffset(1, toVmi.Status.Interfaces[0].IPs).NotTo(BeEmpty())
+func assertICMPPingFail(fromVmi, toVmi *v1.VirtualMachineInstance) {
+	assertIPConnectivityFailed(toVmi, checkICMPPingAndStopOnFailure(fromVmi))
+}
 
-	EventuallyWithOffset(1, func() error {
+func checkICMPPingAndStopOnSucceed(vmi *v1.VirtualMachineInstance) func(string) (bool, error) {
+	return func(ip string) (bool, error) {
 		var err error
-		for _, toIp := range toVmi.Status.Interfaces[0].IPs {
-			if err = tests.PingFromVMConsole(fromVmi, toIp); err == nil {
-				return nil
-			}
+		if err = tests.PingFromVMConsole(vmi, ip); err == nil {
+			return true, nil
 		}
-		return err
-	}, 15*time.Second, time.Second).ShouldNot(Succeed())
+		return false, err
+	}
+}
 
-	ConsistentlyWithOffset(1, func() error {
-		var err error
-		for _, toIp := range toVmi.Status.Interfaces[0].IPs {
-			if err = tests.PingFromVMConsole(fromVmi, toIp); err == nil {
-				return nil
-			}
+func checkICMPPingAndStopOnFailure(vmi *v1.VirtualMachineInstance) func(string) (bool, error) {
```

Comment:
the name is weird/wrong i think,
it doesnt stop, since it doesnt iterate. (only the user of it iterate)
and its rather expect failure instead of the word stop

the opposite happens on checkICMPPingAndStopOnSucceed (but you actually didnt use it)

IMHO i wouldnt do this layering, its most confusing than it was, but up to you and the other reviewers, and for next time if possible please.

Reply:
I am going to revert the commit, after a fresh look it make code more obscure than the stuff we re-use.